library/alloc/src/rc.rs

```diff
@@ -1440,6 +1441,44 @@ impl<T> Rc<[T]> {
         }
     }
 
+    /// Create an `Rc<[T]>` by reusing the underlying memory
+    /// of a `Vec<T>`. This will return the vector if the existing allocation
+    /// is not large enough.
+    #[cfg(not(no_global_oom_handling))]
+    fn try_from_vec_in_place(mut v: Vec<T>) -> Result<Rc<[T]>, Vec<T>> {
+        let layout_elements = Layout::array::<T>(v.len()).unwrap();
+        let layout_allocation = Layout::array::<T>(v.capacity()).unwrap();
+        let layout_rcbox = rcbox_layout_for_value_layout(layout_elements);
+        let mut ptr = NonNull::new(v.as_mut_ptr()).expect("`Vec<T>` stores `NonNull<T>`");
+        if layout_rcbox.size() > layout_allocation.size() {
```

Comment:
This has to test for alignment as well: `Vec<u8>` allocates with byte alignment, but `Rc` requires its allocation to be aligned to `usize`.
```suggestion
        if layout_rcbox.size() > layout_allocation.size() || layout_rcbox.align() > layout_allocation.align() {
```